### PR TITLE
Initial implementation of JAX-RS 3.1 (still using RESTEasy for 3.0) and TCK FAT

### PR DESCRIPTION
**Pull request by andymc12**
- This adds the 3.1 feature code and updates the autofeatures.
- It points to a snapshot build of the JAX-RS 3.1 APIs.
- In includes the snapshot of the TCK running as a FAT.
- All translation files copied over from the 3.0 implementation.
- The implementation is using the 3.0 (transformed) version of RESTEasy.

Part of epic issue #18739.

### CODE REVIEW

**Comment by andymc12** [comment]
#build

**Comment by ramkumar-k-9286** [comment]
It seems like there are no English messages to review in this issue.

Adding the `ID Review not needed` label. If this is not correct please let me know.

Regards,
Ramkumar

**Review by jim-krueger — APPROVED** [review]

**Review by WhiteCat22 — APPROVED** [review]

**Comment by skasund** [comment]
L2 message review not required

**Review by tevans78 — CHANGES_REQUESTED** [review]
Please wait for the feature to at least be prioritzed before merging this.

File: dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restfulWS3.0-xmlBinding3.0.feature [hunk 1]
```diff
@@ -1,7 +1,7 @@
 -include= ~${workspace}/cnf/resources/bnd/feature.props
 symbolicName=io.openliberty.restfulWS3.0-xmlBinding3.0
 visibility=private
-IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.restfulWSClient-3.0))", \
+IBM-Provision-Capability: osgi.identity; filter:="(&(type=osgi.subsystem.feature)(|(osgi.identity=io.openliberty.restfulWSClient-3.0)(osgi.identity=io.openliberty.restfulWSClient-3.1)))", \
  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=io.openliberty.xmlBinding-3.0))"
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restfulWS3.0-xmlBinding3.0.feature]: This PR should go in after https://github.com/OpenLiberty/open-liberty/pull/19534
That will make a `xmlBinding-4.0` feature should be available (even if it is still based on 3.0) and so this auto-feature should take that into account.

File: dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.restfulWS-3.1.feature [hunk 2]
```diff
@@ -0,0 +1,19 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.jakarta.restfulWS-3.1
+visibility=private
+singleton=true
+IBM-App-ForceRestart: uninstall, \
+ install
+IBM-Process-Types: client, \
+ server
+Subsystem-Name: Jakarta RESTful Web Services 3.1
+-features=com.ibm.websphere.appserver.eeCompatible-9.0, \
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/io.openliberty.jakarta.restfulWS-3.1.feature]: Similarly, #19534 will make `eeCompatible-10.0` available

File: dev/io.openliberty.jakarta.rest.3.1.internal_fat_tck/publish/servers/FATServer/server.xml [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        <feature>arquillian-support-jakarta-2.0</feature>
+        <feature>componenttest-2.0</feature>
+        <feature>localConnector-1.0</feature>
+
+        <!--  features under test -->
+        <feature>appSecurity-4.0</feature>
+        <feature>jsonb-2.0</feature>
+        <feature>restfulWS-3.1</feature>
+        <feature>ssl-1.0</feature>
+        <feature>xmlBinding-3.0</feature>
```
Comment (tevans78) [inline dev/io.openliberty.jakarta.rest.3.1.internal_fat_tck/publish/servers/FATServer/server.xml:21]: Use EE 10 features

**Comment by andymc12** [comment]
Last build failed due to missing an Apache HTTP Client 3.1 class - explicitly adding that into the pom.xml - new #build

**Comment by andymc12** [comment]
This currently cannot be merged because the `arquillian-support-jakarta-2.0` feature depends on `servlet-5.0`, and fails when testing with EE10 features. I opened https://github.com/OpenLiberty/liberty-arquillian/pull/114 which would resolve issue https://github.com/OpenLiberty/liberty-arquillian/issues/113 - once that issue is resolved and pulled in to Liberty, then development on this PR could continue.

**Comment by jim-krueger** [comment]
Closing,  changes updated and merged under #20505

**Pull request closed without merging**